File: domain/src/main/java/com/aceplus/domain/model/sale/salereturn/SaleReturn.java
package com.aceplus.domain.model.sale.salereturn;

import java.io.Serializable;

/**
 * Created by yma on 2/16/17.
 *
 * SaleReturn
 */

public class SaleReturn implements Cloneable, Serializable {

    /**
     * sale return id
     */
    private String saleReturnId;

    /**
     * customer id
     */
    private int customerId;

    /**
     * location id
     */
    private int locationId;

    /**
     * net amount
     */
    private double amt;

    /**
     * return cash amount
     */
    private double payAmt;

    /**
     * return dis amount
     */
    private double disAmt;

    /**
     * return tax amount
     */
    private double taxAmt;

    /**
     * return tax percent
     */
    private double taxPercent;

    /**
     * pc address
     */
    private String pcAddress;

    /**
     * return date
     */
    private String returnedDate;

    /**
     * Invoice status
     */
    private String invoiceStatus;

    /**
     * Sale Man Id
     */
    private String saleManId;

    /**
     * Sale id
     */
    private String saleId;

    /**
     * Discount of sale return
     */
    private double discountAmount;

    /**
     * Getter method for saleReturnId
     *
     * @return saleReturnId
     */
    public String getSaleReturnId() {
        return saleReturnId;
    }

    /**
     * Setter method for saleReturnId
     *
     * @param saleReturnId sale return id
     */
    public void setSaleReturnId(String saleReturnId) {
        this.saleReturnId = saleReturnId;
    }

    /**
     * Getter method for customerId
     *
     * @return customerId
     */
    public int getCustomerId() {
        return customerId;
    }

    /**
     * Setter method for customerId
     *
     * @param customerId customer id
     */
    public void setCustomerId(int customerId) {
        this.customerId = customerId;
    }

    /**
     * Getter method for locationId
     *
     * @return locationId
     */
    public int getLocationId() {
        return locationId;
    }

    /**
     * Setter method for locationId
     *
     * @param locationId location id
     */
    public void setLocationId(int locationId) {
        this.locationId = locationId;
    }

    /**
     * Getter method for amt
     *
     * @return amt
     */
    public double getAmt() {
        return amt;
    }

    /**
     * Setter method for amt
     *
     * @param amt net amount
     */
    public void setAmt(double amt) {
        this.amt = amt;
    }

    /**
     * Getter method for payAmt
     *
     * @return payAmt
     */
    public double getPayAmt() {
        return payAmt;
    }

    /**
     * Setter method for payAmt
     *
     * @param payAmt returned cash amount
     */
    public void setPayAmt(double payAmt) {
        this.payAmt = payAmt;
    }

    /**
     * Getter method for pcAddress
     *
     * @return pcAddress
     */
    public String getPcAddress() {
        return pcAddress;
    }

    /**
     * Setter method for pcAddress
     *
     * @param pcAddress pc address
     */
    public void setPcAddress(String pcAddress) {
        this.pcAddress = pcAddress;
    }

    /**
     * Getter method for returnedDate
     *
     * @return returnedDate
     */
    public String getReturnedDate() {
        return returnedDate;
    }

    /**
     * Setter method for returnedDate
     *
     * @param returnedDate return date
     */
    public void setReturnedDate(String returnedDate) {
        this.returnedDate = returnedDate;
    }

    /**
     * Getter method for invoice status
     *
     * @return invoiceStatus
     */
    public String getInvoiceStatus() {
        return invoiceStatus;
    }

    /**
     * Setter invoice status
     *
     * @param invoiceStatus credit or cash
     */
    public void setInvoiceStatus(String invoiceStatus) {
        this.invoiceStatus = invoiceStatus;
    }

    /**
     * Getter method for sale man id
     *
     * @return sale man id
     */
    public String getSaleManId() {
        return saleManId;
    }

    /**
     * Setter method for sale man id.
     *
     * @param saleManId sale man id
     */
    public void setSaleManId(String saleManId) {
        this.saleManId = saleManId;
    }

    /**
     * Getter method for sale id
     *
     * @return saleId
     */
    public String getSaleId() {
        return saleId;
    }

    /**
     * Setter method for sale id
     *
     * @param saleId sale id
     */
    public void setSaleId(String saleId) {
        this.saleId = saleId;
    }

    /**
     * getter method of dis Amount @return disAmt
     */
    public double getDisAmt() {
        return disAmt;
    }
    /**
     * setter method of dis Amount @param disAmt
     */
    public void setDisAmt(double disAmt){
        this.disAmt = disAmt;
    }
    /**
     * getter method of tax Amount @return taxAmt
     */
    public double getTaxAmt(){
        return taxAmt;
    }
    /**
     * setter method of tax Amount @param taxAmt
     */
    public void setTaxAmt(double taxAmt){
        this.taxAmt = taxAmt;
    }
    /**
     * getter method of tax Percent @return taxPercent
     */
    public double getTaxPercent(){
        return taxPercent;
    }
    /**
     * setter method of tax Percent @param taxPercent
     */
    public void setTaxPercent(double taxPercent){
        this.taxPercent = taxPercent;
    }
}

File: domain/src/main/java/com/aceplus/domain/entity/sale/salereturn/SaleReturn.kt
package com.aceplus.domain.entity.sale.salereturn

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


@Entity(tableName = "sale_return")
class SaleReturn {

    @PrimaryKey
    @ColumnInfo(name = "sale_return_id")
    @SerializedName("sale_return_id")
    @Expose
    var sale_return_id: Int = 0

    @ColumnInfo(name = "customer_id")
    @SerializedName("customer_id")
    @Expose
    var customer_id: Int = 0

    @ColumnInfo(name = "location_id")
    @SerializedName("location_id")
    @Expose
    var location_id: Int = 0

    @ColumnInfo(name = "amount")
    @SerializedName("amount")
    @Expose
    var amount: Double = 0.0

    @ColumnInfo(name = "pay_amount")
    @SerializedName("pay_amount")
    @Expose
    var pay_amount: Double = 0.0

    @ColumnInfo(name = "pc_address")
    @SerializedName("pc_address")
    @Expose
    var pc_address: String? = null

    @ColumnInfo(name = "return_date")
    @SerializedName("return_date")
    @Expose
    var return_date: String? = null

    @ColumnInfo(name = "rate")
    @SerializedName("rate")
    @Expose
    var rate: String? = null

    @ColumnInfo(name = "currency_id")
    @SerializedName("currency_id")
    @Expose
    var currency_id: Int = 0

    @ColumnInfo(name = "invoice_status")
    @SerializedName("invoice_status")
    @Expose
    var invoice_status: String? = null

    @ColumnInfo(name = "sale_man_id")
    @SerializedName("sale_man_id")
    @Expose
    var sale_man_id: Int = 0

    @ColumnInfo(name = "sale_id")
    @SerializedName("sale_id")
    @Expose
    var sale_id: Int = 0

    @ColumnInfo(name = "discount")
    @SerializedName("discount")
    @Expose
    var discount: Double = 0.0

    @ColumnInfo(name = "tax")
    @SerializedName("tax")
    @Expose
    var tax: Double = 0.0

    @ColumnInfo(name = "tax_percent")
    @SerializedName("tax_percent")
    @Expose
    var tax_percent: Double = 0.0

    @ColumnInfo(name = "delete_flag")
    @SerializedName("delete_flag")
    @Expose
    var delete_flag: String? = null
}